### ISSUE TITLE
Fetching from origin makes GH Desktop think all files in a branch are new

### ISSUE THREAD

**Issue by rheath-dremio** [desktop/desktop#13126]
### Describe the bug
Whichever branch I'm in, including my master branch, clicking the **Fetch origin** button causes GH Desktop to put all of the files in my branch in the **Changes** pane and mark all of the files as new.

### Version & OS

Version 2.9.4 (x64) of GitHub Desktop
MacOS 11.6

### Steps to reproduce the behavior

See the description.

### Expected behavior

Fetching files from origin for a branch should not make GH Desktop think all files in the branch are new.

### Actual behavior

See the description.

### Logs

2021-10-14T15:37:47.219Z - info: [ui] [AppStore] loading 2 repositories from store
2021-10-14T15:37:47.221Z - info: [ui] [AppStore] found account: rheath-dremio (rheath-dremio)
2021-10-14T15:37:48.139Z - info: [ui] launching: 2.9.4 (Mac OS 11.6.0)
2021-10-14T15:37:48.140Z - info: [ui] execPath: '/Applications/GitHub Desktop.app/Contents/Frameworks/GitHub Desktop Helper (Renderer).app/Contents/MacOS/GitHub Desktop Helper (Renderer)'
2021-10-14T15:37:48.543Z - info: [ui] Stats reported.
2021-10-14T15:37:51.178Z - info: [ui] Executing fetch: git -c credential.helper= -c protocol.version=2 fetch --progress --prune origin (took 1.200s)
2021-10-14T16:38:30.139Z - info: [ui] Executing fetch: git -c credential.helper= -c protocol.version=2 fetch --progress --prune origin (took 1.420s)
2021-10-14T17:25:36.505Z - info: [ui] [Timing] Action 'checkout branch from list' for 'dremio/documentation' took 0.000s
2021-10-14T17:25:40.097Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: documentation - rheath-dremio (has token)
2021-10-14T17:25:40.296Z - error: [ui] Could not load new contents for syntax highlighting
Error: ENOENT: no such file or directory, open '/Users/robert.heath/Documents/GitHub/documentation/.DS_Store'
2021-10-14T17:25:40.414Z - info: [ui] [Timing] Action 'stash and checkout' for 'dremio/documentation' took 0.306s
2021-10-14T17:25:42.294Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: documentation - rheath-dremio (has token)
2021-10-14T17:25:43.458Z - info: [ui] Executing fetch: git -c credential.helper= -c protocol.version=2 fetch --progress --prune origin (took 1.146s)

### Additional context

This bug makes GH Desktop unusable.

**Comment by steveward** [comment]
Thanks for the report @rheath-dremio. Can you open the repository in the command line (`Repository` > `Open in Terminal`) and share the output of `git diff <file>`? You can just choose one file with a change -- my hunch is this is either a file mode or line endings issue and hopefully that will confirm it.

**Comment by rheath-dremio** [comment]
Thanks for getting back to me. It turns out that the problem is not GH Desktop. Somehow, the .gitignore file for the repo was deleted, or its content was changed. We have a script that generates HTML from our Markdown (the repo is for documentation), and the script creates a folder that was not being ignored. When I saw the 2000+ files in the Changes pane of GH Desktop, I didn't notice the path for the files began with the name of that folder. I fixed the .gitignore file, pushed it, ran the script, and saw that the files no longer appear in the Changes pane.

Chalk this issue up to a user error.
